agent/agentcontainers/containers_dockercli.go

```diff
@@ -31,6 +34,211 @@ func NewDocker(execer agentexec.Execer) Lister {
 	}
 }
 
+// ContainerEnvInfoer is an implementation of agentssh.EnvInfoer that returns
+// information about a container.
+type ContainerEnvInfoer struct {
+	container string
+	user      *user.User
+	userShell string
+	env       []string
+}
+
+// EnvInfo returns information about the environment of a container.
+func EnvInfo(ctx context.Context, execer agentexec.Execer, container, containerUser string) (*ContainerEnvInfoer, error) {
```

Comment:
Where will this be called? Right now it's only used in tests.

Reply:
See here: https://github.com/coder/coder/pull/16638